src/backend/joanie/signature/backends/base.py

```diff
@@ -42,27 +42,49 @@ def get_setting(self, name):
 
         return getattr(settings, setting_name, None)
 
-    def confirm_signature(self, reference):
+    def confirm_student_signature(self, reference):
         """
         Update the contract object when the file has been signed with the signature provider.
         We verify if the contract is still in its validity period to be sign, and if it's True
-        we update the field 'signed_on' with a new timestamp.
+        we update the field 'student_signed_on' with a new timestamp.
         """
         contract = models.Contract.objects.get(signature_backend_reference=reference)
 
         if not contract.is_eligible_for_signing():
             logger.error(
-                "Contract's validity date has passed for contract id : '%s'",
+                "When student signed, contract's validity date has passed for contract id : '%s'",
+                contract.id,
+            )
+            raise ValidationError(
+                "The contract validity date of expiration has passed."
+            )
+
+        contract.student_signed_on = django_timezone.now()
+        contract.save()
+        logger.info("Student signed the contract '%s'", contract.id)
```

Comment:
On rebase, keep in mind to keep `contract.order.enroll_user_to_course_run()` call on student signature